iothub_client/inc/iothub_device_client_ll.h

```diff
@@ -374,6 +374,108 @@ typedef struct IOTHUB_CLIENT_CORE_LL_HANDLE_DATA_TAG* IOTHUB_DEVICE_CLIENT_LL_HA
+                        void* userContextCallback);
+/**
+* @brief    Sends properties to IoT Hub
+*
+* @param[in]    iotHubClientHandle           The handle created by a call to the create function.
+* @param[in]    properties                   Serialized property data to be sent to IoT Hub.  This buffer can either be
+*                                            manually serialized created with IoTHubClient_Serialize_ReportedProperties 
+*                                            or IoTHubClient_Serialize_WritablePropertyResponse.
+* @param[in]    propertiesLength             Number of bytes in the properties buffer.
+* @param[in]    propertyAcknowledgedCallback Optional callback specified by the application to be called with the
+*                                            result of the transaction.
+* @param[in]    userContextCallback          Optional User specified context that will be provided to the callback.
+*
+* @remarks
+*            The application behavior is undefined if the user calls
+*            the IoTHubDeviceClient_LL_Destroy function from within any callback.
+*
+* @return    IOTHUB_CLIENT_OK upon success or an error code upon failure.
+*/
+IOTHUB_CLIENT_RESULT IoTHubDeviceClient_LL_SendPropertiesAsync(
+                         IOTHUB_DEVICE_CLIENT_LL_HANDLE iotHubClientHandle, 
+                         const unsigned char* properties,
+                         size_t propertiesLength,
+                         IOTHUB_CLIENT_PROPERTY_ACKNOWLEDGED_CALLBACK propertyAcknowledgedCallback,
+                         void* userContextCallback);
+/**
+* @brief   Retrieves all properties from IoT Hub.
+*
+* @param[in]   iotHubClientHandle    The handle created by a call to the create function.
+* @param[in]   propertyCallback      Callback invoked when properties are retrieved.
+                                     The API IoTHubClient_Deserialize_Properties can help deserialize the raw payload stream.
+* @param[in]   userContextCallback   Optional user specified context that will be provided to the
+*                                    callback.
+*
+* @remarks
+*           The application behavior is undefined if the user calls
+*           the IoTHubDeviceClient_LL_Destroy function from within any callback.
+*
+* @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
+*/
+IOTHUB_CLIENT_RESULT IoTHubDeviceClient_LL_GetPropertiesAsync(
+                        IOTHUB_DEVICE_CLIENT_LL_HANDLE iotHubClientHandle, 
+                        IOTHUB_CLIENT_PROPERTIES_RECEIVED_CALLBACK propertyCallback, 
+                        void* userContextCallback);
+
+/**
+* @brief   Retrieves all properties from IoT Hub and also subscribes for updates to writable properties.
+*
+* @param[in]   iotHubClientHandle      The handle created by a call to the create function.
+* @param[in]   propertyUpdateCallback  Callback both on initial retrieval of properties stored on IoT Hub
```

Comment:
I'm wondering in general how many characters we are allowing on each line.  Sometimes lines go on for ever, but here you are clearly moving characters to a new line.  Not a huge deal, but even within this PR the formatting is inconsistent.

Reply:
I've brought it to about 120 chars / line max for new stuff.  80 I believe is embedded but the existing C SDK goes way past 80 as-is.

Reply:
sounds good. Yeah, I've noticed the C SDK is all over the place, which is why I'm not too concerned about it. Thanks.